otter/convergence/model.py

```diff
@@ -126,6 +154,69 @@ def __init__(self):
         assert self.state in ServerState.iterconstants(), \
             "%r is not a ServerState" % (self.state,)
 
+    @classmethod
+    def group_id_from_metadata(cls, metadata):
+        """
+        Get the group ID of a server based on the metadata.
+
+        The old key was ``rax:auto_scaling_group_id``, but the new key is
+        ``rax:autoscale:group:id``.  Try pulling from the new key first, and
+        if it doesn't exist, pull from the old.
+        """
+        return metadata.get(
+            "rax:autoscale:group:id",
+            metadata.get("rax:auto_scaling_group_id", None))
+
+    @classmethod
+    def lbs_from_metadata(cls, metadata):
+        """
+        Get the desired load balancer descriptions based on the metadata.
+
+        :return: ``dict`` of `ILBDescription` providers
+        """
+        lbs = get_service_metadata('autoscale', metadata).get('lb', {})
+        desired_lbs = {}
+
+        for lb_id, v in lbs.get('CloudLoadBalancer', {}).iteritems():
+            # if malformed, skiped the whole key
+            try:
+                configs = json.loads(v)
+                if isinstance(configs, list):
+                    desired_lbs[lb_id] = [
+                        CLBDescription(lb_id=lb_id, port=c['port'])
+                        for c in configs]
+            except (ValueError, KeyError, TypeError):
+                pass
+
+        return pmap(desired_lbs)
+
+    @classmethod
+    def generate_metadata(cls, group_id, lb_descriptions):
+        """
+        Generate autoscale-specific Nova server metadata given the group ID and
+        an iterable of :class:`ILBDescription` providers.
+
+        NOTE: Currently this ignores RCv3 settings and draining timeout
+        settings, since they haven't been implemented yet.
+
+        :return: a metadata `dict` containing the group ID and LB information
+        """
+        metadata = {
+            'rax:auto_scaling_group_id': group_id,
```

Comment:
I'm not really sure why we want to write both these keys. If we're writing the new ID, then all of our services should know how to look for the new keys. Is it just for backwards compatibility with Reach? Can't we just get them to start looking for the new keys before we deploy convergence?